### PR TITLE
CP-15244: Add OS type check and IP Parse

### PR DESCRIPTION
**Pull request by cheng-z**
Signed-off-by: chengz <cheng.zhang@citrix.com>

### CODE REVIEW

**Comment by cheng-z** [comment]
Hi @phusl 
Could you please let me know if you have any comments?
Thanks.

File: feature/FeatureIPSetting.go [hunk 1]
```diff
@@ -86,6 +91,158 @@ func (f *FeatureIPSetting) GetChildrens(key string) []string {
...
 
+type OSType int
+
+const (
+	OTHER  OSType = 0
+	CENTOS OSType = 1
+)
+
+func GetCurrentOSType() OSType {
+	distributionFile, err := os.OpenFile("/var/cache/xe-linux-distribution", os.O_RDONLY, 0666)
```
Comment (phusl) [inline feature/FeatureIPSetting.go]: Seems that we shall `run it` with shell rather than `read it`.

File: feature/FeatureIPSetting.go [hunk 2]
```diff
@@ -86,6 +91,158 @@ func (f *FeatureIPSetting) GetChildrens(key string) []string {
...
+
+func AddSettingHistory(conf Configuration) {
+	configurations := GetSettingHistory()
+	found := false
+	for _, configuration := range configurations {
+		if configuration.Mac == conf.Mac && configuration.IsIPv6 == conf.IsIPv6 {
+			found = true
+		}
+	}
+	if found == false {
```
Comment (phusl) [inline feature/FeatureIPSetting.go]: woud be better `if !found`

File: feature/FeatureIPSetting.go [hunk 3]
```diff
@@ -86,6 +91,158 @@ func (f *FeatureIPSetting) GetChildrens(key string) []string {
...
+			encoder := json.NewEncoder(file)
+			encoder.Encode(configurations)
+		}
+	}
+}
+
+func GetHistoryByMac(mac string, isIPv6 bool) Configuration {
+	configurations := GetSettingHistory()
+	for _, configuration := range configurations {
+		if configuration.Mac == mac && configuration.IsIPv6 == isIPv6 {
```
Comment (phusl) [inline feature/FeatureIPSetting.go]: would be better `configuration.IsIPv6`

File: feature/FeatureIPSetting.go [hunk 4]
```diff
@@ -86,6 +91,158 @@ func (f *FeatureIPSetting) GetChildrens(key string) []string {
...
+	}
+	return Configuration{}
+}
+
+func RemoveHistoryByMac(mac string, isIPv6 bool) {
+	configurations := GetSettingHistory()
+	found := false
+	index := 0
+	for i, configuration := range configurations {
+		if configuration.Mac == mac && configuration.IsIPv6 == isIPv6 {
```
Comment (phusl) [inline feature/FeatureIPSetting.go]: would be better `configuration.IsIPv6`

**Comment by cheng-z** [comment]
Hi @phusl 
Please see my change in latest commit as: 0602528c37dfa0fdf90e85584b3f521585354de4
Thanks.

**Pull request merged**